cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -118,6 +122,57 @@ class CordappPlugin : Plugin<Project> {
         return filteredDeps.toUniqueFiles(runtimeConfiguration) - excludeDeps.toUniqueFiles(runtimeConfiguration)
     }
 
+    private fun getAndCheckVersionInfo(project: Project): Pair<Int?, Int?> {
+        // If the target version is not set, try to determine the platformversion of the Corda jar this project
+        // depends on and use that.
+        val targetVersion: Int? =
+                if (cordapp.info?.targetVersion == null) {
+                    project.logger.warn("No target version specified. Attempting to determine platform version from the project's Corda dependency.")
+                    val platformVersion = getPlatformVersion(project)
+                    if (platformVersion == null) {
+                        project.logger.warn("Could not determine platform version from the project's Corda dependency.")
+                    } else {
+                        project.logger.info("Setting target version to ${platformVersion}.")
+                    }
+                    platformVersion
+                } else {
+                    cordapp.info?.targetVersion
+                }
+
+        // If the minimum platform version is not set, default to 1.
+        val minPlatformVersion: Int = cordapp.info?.minPlatformVersion ?: 1
+        check(targetVersion != null) { "Target version was not set and could not be determined from the project's Corda dependency. Please specify the target version of your CorDapp." }
+        check(targetVersion!! >= 1) { "Target version must not be smaller than 1." }
+        check(targetVersion >= minPlatformVersion) { "Target version must not be smaller than min platform version." }
+
+        return Pair(targetVersion, minPlatformVersion)
+    }
+
+    /**
+     * Retrieve the Corda platform version from the project's Corda jar (if present)
+     *
+     * @param project The project environment this plugin executes in.
+     * @return The platform version of the projects Corda jar, or null if it cannot be determined.
+     */
+    private fun getPlatformVersion(project: Project): Int? {
+        val releaseVersion =
+                try {
+                    project.rootProject.ext<String>("corda_release_version")
+                } catch (e: ExtraPropertiesExtension.UnknownPropertyException) {
+                    project.logger.warn(e.message)
+                    null
+                } ?: return null
+        val maybeJar = project.configuration("runtime").filter {
+            "corda-$releaseVersion.jar" in it.toString() || "corda-enterprise-$releaseVersion.jar" in it.toString()
+        }
```

Comment:
Does code for locating `corda.jar` already exist somewhere? I thought it did...

Reply:
There's something similar in `verifyAndGetRuntimeJar`, but it doesn't do exactly what I want